File: Postie Notes/PostieSettings.vb
Public Class PostieSettings
    Private Sub PostieSettings_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        DSLText.Text = My.Settings.DefaultSaveDirectory
    End Sub

    Private Sub PostieSettings_FormClosing(sender As Object, e As FormClosingEventArgs) Handles MyBase.FormClosing
        My.Settings.DefaultSaveDirectory = DSLText.Text
        My.Settings.Save()
    End Sub

    Private Sub DSLBrowseButton_Click(sender As Object, e As EventArgs) Handles DSLBrowseButton.Click
        Dim FolderDialogue As New FolderBrowserDialog
        FolderDialogue.ShowNewFolderButton = True
        If (FolderDialogue.ShowDialog() = DialogResult.OK) Then
            DSLText.Text = FolderDialogue.SelectedPath
        End If
    End Sub
End Class

File: Postie Notes/PostieSettings.cs
using System;
//using System.Collections.Generic;
//using System.ComponentModel;
//using System.Data;
//using System.Drawing;
//using System.Text;
using System.Windows.Forms;

namespace Postie_Notes
{
    public partial class PostieSettings : Form
    {
        public PostieSettings()
        {
            InitializeComponent();
        }

        private void DSLBrowseButton_Click(object sender, EventArgs e)
        {
            FolderBrowserDialog FolderDialogue = new FolderBrowserDialog();
            FolderDialogue.ShowNewFolderButton = true;
            if (FolderDialogue.ShowDialog() == DialogResult.OK)
            {
                DSLText.Text = FolderDialogue.SelectedPath;
            }
        }

        private void PostieSettings_Load(object sender, EventArgs e)
        {
            DSLText.Text = Properties.Settings.Default.DefaultSaveDirectory;
        }

        private void PostieSettings_FormClosing(object sender, FormClosingEventArgs e)
        {
            Properties.Settings.Default.DefaultSaveDirectory = DSLText.Text;
            Properties.Settings.Default.Save();
        }
    }
}
